android/app/src/main/res/layout/fragment_osm_login_bottom.xml

```diff
@@ -0,0 +1,108 @@
+    <LinearLayout
+      android:layout_width="match_parent"
+      android:layout_height="match_parent"
+      android:clipChildren="false"
+      android:clipToPadding="false"
+      android:orientation="vertical"
+      android:padding="@dimen/margin_base"
+      tools:ignore="ScrollViewSize">
+      <LinearLayout
+          android:id="@+id/login_bottom_sheet"
+          android:layout_width="match_parent"
+          android:layout_height="wrap_content"
+          android:orientation="vertical">
+
+        <com.google.android.material.textfield.TextInputLayout
+            style="@style/MwmWidget.Editor.CustomTextInput"
+            android:layout_marginTop="@dimen/margin_half"
+            android:textColorHint="?android:textColorSecondary"
+            app:endIconMode="custom">
+          <com.google.android.material.textfield.TextInputEditText
+              android:id="@+id/osm_username"
+              style="@style/MwmWidget.Editor.FieldLayout.EditText"
+              android:padding="@dimen/margin_half_double_plus"
+              android:autofillHints="emailAddress"
+              android:hint="@string/email_or_username" />
+        </com.google.android.material.textfield.TextInputLayout>
+        <com.google.android.material.textfield.TextInputLayout
+            style="@style/MwmWidget.Editor.CustomTextInput"
+            android:textColorHint="?android:textColorSecondary"
+            app:endIconMode="password_toggle"
+            app:endIconTint="?android:textColorSecondary">
+          <com.google.android.material.textfield.TextInputEditText
+              android:id="@+id/osm_password"
+              style="@style/MwmWidget.Editor.FieldLayout.EditText"
+              android:padding="@dimen/margin_half_double_plus"
+              android:autofillHints="password"
+              android:hint="@string/password"
+              android:inputType="textPassword" />
+        </com.google.android.material.textfield.TextInputLayout>
+      </LinearLayout>
+
+      <FrameLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/margin_base">
+        <Button
+          android:id="@+id/login"
+          style="@style/MwmWidget.Button.Accent"
+          android:layout_width="match_parent"
+          android:layout_height="wrap_content"
+          android:text="Login with username"
+            android:layout_marginTop="@dimen/margin_half"
+          android:textAppearance="@style/MwmTextAppearance.Body2.Light" />
+        <ProgressBar
+          android:id="@+id/osm_login_progress"
+          android:layout_width="@dimen/editor_auth_btn_height"
+          android:layout_height="@dimen/editor_auth_btn_height"
+          android:layout_gravity="center"
+          android:elevation="@dimen/design_fab_elevation"
+          android:visibility="gone" />
+      </FrameLayout>
+
+      <LinearLayout
+          android:layout_width="match_parent"
```

Comment:
For some reason, formatting is inconsistent.

Reply:
Fixed formatting